### ISSUE TITLE
Unique IDs for game npcs, entities, players, items, etc

### ISSUE THREAD

**Issue by bitcraft** [Tuxemon/Tuxemon#669]
We currently require all game npcs to be unique and have a json file associated with them.  Their ID is the slug, and when using scripts, we refer to npc entities by their slug.  In most cases this works well.  However, we may find that we need to create npcs, or map objects (entites) and later refer to them.  I propose we have a unique ID assigned to all objects when they are created.  This ID can be used in place of a slug.

**Comment by Qiangong2** [comment]
why not just refer it by its slug the entire time?

**Comment by xirsoi** [comment]
This issue is especially relevant to Tuxemon, as you say, because the slug is really only referring to the species, and we currently have no way to identify an individual Tuxemon except in relation to some other game object (like referring to the nth monster in the player's party).

**Comment by bitcraft** [comment]
yeah, exactly.  the txmn networking code, i believe, does use uuid's for objects, so maybe that is the way to go.  if a mapper wants to, they can store the id in game variable for easier reference later.

**Comment by bitcraft** [comment]
this has been implemented for monsters
